packages/js/product-editor/src/blocks/shipping-fee/edit.tsx

```diff
@@ -0,0 +1,194 @@
+/**
+ * External dependencies
+ */
+import { BlockEditProps } from '@wordpress/blocks';
+import { Link } from '@woocommerce/components';
+import {
+	EXPERIMENTAL_PRODUCT_SHIPPING_CLASSES_STORE_NAME,
+	ProductShippingClass,
+} from '@woocommerce/data';
+import { getNewPath } from '@woocommerce/navigation';
+import { recordEvent } from '@woocommerce/tracks';
+import { useBlockProps } from '@wordpress/block-editor';
+import {
+	BaseControl,
+	RadioControl,
+	SelectControl,
+} from '@wordpress/components';
+import { useInstanceId } from '@wordpress/compose';
+import { useEntityProp } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+import {
+	Fragment,
+	createElement,
+	createInterpolateElement,
+	useEffect,
+	useState,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { ShippingFeeBlockAttributes } from './types';
+import { useValidation } from '../../hooks/use-validation';
+
+const FOLLOW_CLASS_OPTION_VALUE = 'follow_class';
+const FREE_SHIPPING_OPTION_VALUE = 'free_shipping';
+
+const options = [
+	{
+		label: __( 'Follow class', 'woocommerce' ),
+		value: FOLLOW_CLASS_OPTION_VALUE,
+	},
+	{
+		label: __( 'Free shipping', 'woocommerce' ),
+		value: FREE_SHIPPING_OPTION_VALUE,
+	},
+];
+
+export function Edit( {
+	attributes,
+}: BlockEditProps< ShippingFeeBlockAttributes > ) {
+	const { title } = attributes;
+
+	const blockProps = useBlockProps( {
+		className: 'wp-block-woocommerce-product-shipping-fee-fields',
```

Comment:
Same question as other PR regarding manually overriding class names.

Reply:
see -> https://github.com/woocommerce/woocommerce/pull/37683#discussion_r1164545010

Reply:
Responded [here](https://github.com/woocommerce/woocommerce/pull/37683#discussion_r1166190568).  Please let me know if I misunderstood or if there's a different reason we need to add the class name.

Reply:
Thanks, done -> https://github.com/woocommerce/woocommerce/pull/37642/commits/0036e1ebcca4749ea7582ad8e00e526e10a0bc1b